src/Processors/Formats/Impl/PrettyBlockOutputFormat.cpp

```diff
@@ -142,17 +141,38 @@ GridSymbols ascii_grid_symbols {
 
 }
 
-
 void PrettyBlockOutputFormat::write(Chunk chunk, PortKind port_kind)
 {
-    UInt64 max_rows = format_settings.pretty.max_rows;
-
-    if (total_rows >= max_rows)
+    if (total_rows >= format_settings.pretty.max_rows)
     {
         total_rows += chunk.getNumRows();
         return;
     }
+    if (mono_block)
+    {
+        if (port_kind == PortKind::Main)
+        {
+            if (!mono_chunk)
+            {
+                mono_chunk = std::move(chunk);
+                return;
+            }
+
+            mono_chunk.append(chunk);
+            return;
+        }
+        else
+        {
+            /// Should be written from writeSuffix()
+            assert(!mono_chunk);
+        }
```

Comment:
Just style, applying up to you.
Already have a return in the main if branch, but not a big deal. Also, I haven't checked if this snippet built

```suggestion
        if (port_kind == PortKind::Main)
        {
            if (mono_chunk)
                mono_chunk.append(chunk);
            else
                mono_chunk = std::move(chunk);
            return;
        }
        
        /// Should be written from writeSuffix()
        assert(!mono_chunk);
```